lib/pos_mempool.go

```diff
@@ -241,11 +270,72 @@ func (mp *PosMempool) Start() error {
 			return errors.Wrapf(err, "PosMempool.Start: Problem loading persisted transactions")
 		}
 	}
+	mp.startAugmentedViewRefreshRoutine()
 
 	mp.status = PosMempoolStatusRunning
 	return nil
 }
 
+func (mp *PosMempool) startAugmentedViewRefreshRoutine() {
+
+	go func() {
+	out:
+		for {
+			select {
+			case <-time.After(time.Duration(mp.augmentedBlockViewRefreshIntervalMillis) * time.Millisecond):
+				// If we're not within 10 blocks of the PoS cutover, we don't need to update the
+				// augmentedLatestBlockView.
+				if mp.latestBlockHeight+10 < uint64(mp.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+					continue
+				}
+				// Update the augmentedLatestBlockView with the latest block view.
+				mp.RLock()
+				newView, err := mp.readOnlyLatestBlockView.CopyUtxoView()
+				if err != nil {
+					glog.Errorf("PosMempool.startAugmentedViewRefreshRoutine: Problem copying utxo view: %v", err)
+					continue
+				}
+				mp.RUnlock()
+				for _, txn := range mp.GetTransactions() {
+					copiedView, err := newView.CopyUtxoView()
+					if err != nil {
+						glog.Errorf("PosMempool.startAugmentedViewRefreshRoutine: Problem copying utxo view: %v", err)
+						continue
+					}
+					_, _, _, _, err = copiedView.ConnectTransaction(
+						txn.GetTxn(), txn.Hash(), 0, uint32(mp.latestBlockHeight)+1, time.Now().UnixNano(), false,
+						false)
+					// If the transaction successfully connects, we set the newView to the copiedView
+					// and proceed to the next transaction.
+					if err == nil {
+						newView = copiedView
+						continue
+					}
+					// If the transaction failed to connect, we connect the transaction as a failed txn
+					// directly on newView.
+					if mp.latestBlockHeight+1 >= uint64(mp.params.ForkHeights.ProofOfStake2ConsensusCutoverBlockHeight) {
+						// Try to connect as failing txn directly to newView
+						_, _, _, err = newView._connectFailingTransaction(
+							txn.GetTxn(), uint32(mp.latestBlockHeight), false)
+						if err != nil {
+							glog.Errorf(
+								"PosMempool.startAugmentedViewRefreshRoutine: Problem connecting transaction: %v", err)
+						}
+					}
+				}
+				// Grab the augmentedLatestBlockViewMutex write lock and update the augmentedLatestBlockView.
+				mp.augmentedLatestBlockViewMutex.Lock()
+				mp.augmentedLatestBlockView = newView
+				mp.augmentedLatestBlockViewMutex.Unlock()
+				// Increment the augmentedLatestBlockViewSequenceNumber.
+				atomic.AddInt64(&mp.augmentedLatestBlockViewSequenceNumber, 1)
+			case <-mp.quit:
+				break out
```

Comment:
nit: you can just just return, no?
```suggestion
				return
```